google/cloud/spanner/transaction.cc

```diff
@@ -86,6 +95,12 @@ Transaction::Transaction(ReadWriteOptions opts) {
   impl_ = std::make_shared<internal::TransactionImpl>(std::move(selector));
 }
 
+Transaction::Transaction(PartitionedDmlOptions opts) {
+  google::spanner::v1::TransactionSelector selector;
+  *selector.mutable_begin() = MakeOpts(std::move(opts.pd_opts_));
```

Comment:
I think we should consider keeping "partitioned dml" stuff out of the Transaction classess and the `transaction.*` files. Instead, I think we can simply set the TransactionSelector proto directly in the implementation in `connection_impl.cc`. This would make it so the only place "partitioned dml" is uttered in our user-facing code is in the one function `Client::ExecutePartitionedDml`, which does not even take a `Transaction` nor options. I think the rest of the plumbing provided by the `Transaction` class is not needed for PartitionedDml transactions because they are never exposed to users, are not re-usable, and cannot be committed.

Reply:
Done.